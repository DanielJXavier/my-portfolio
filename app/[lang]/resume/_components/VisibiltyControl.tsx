import { ReactNode, useState } from "react";

export default function VisibilityControl({
  children,
  className,
  defaultVisible,
}: {
  children: ReactNode;
  className?: string;
  defaultVisible?: boolean;
}) {
  const [visible, setVisible] = useState(defaultVisible ?? true);

  return (
    <li
      className={`${className ?? ""} flex gap-x-1.5 ${
        !visible ? "opacity-50 line-through print:hidden" : ""
      }`}
      data-testid="visibility-control"
      {...(visible ? { "data-visible": "" } : {})}
    >
      <input
        className="print:hidden"
        type="checkbox"
        checked={visible}
        onChange={() => setVisible(!visible)}
      />
      {children}
    </li>
  );
}
