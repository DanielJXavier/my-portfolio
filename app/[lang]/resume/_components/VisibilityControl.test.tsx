import "@testing-library/jest-dom";

import { fireEvent, render, screen } from "@testing-library/react";

import VisibiltyControl from "./VisibiltyControl";

describe("VisibiltyControl component", () => {
  it("renders the component", () => {
    const { container } = render(<VisibiltyControl>children</VisibiltyControl>);

    expect(container).toMatchSnapshot();
  });

  it("hide the children when printing", () => {
    render(<VisibiltyControl>children</VisibiltyControl>);

    const visibilityControl = screen.getByTestId("visibility-control");

    fireEvent.click(visibilityControl.childNodes[0]);

    expect(visibilityControl).toHaveClass(
      "opacity-50 line-through print:hidden"
    );
  });
});
