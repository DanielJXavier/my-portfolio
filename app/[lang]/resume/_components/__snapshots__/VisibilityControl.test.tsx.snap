// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisibiltyControl component renders the component 1`] = `
<div>
  <li
    class=" flex gap-x-1.5 "
    data-testid="visibility-control"
    data-visible=""
  >
    <input
      checked=""
      class="print:hidden"
      type="checkbox"
    />
    children
  </li>
</div>
`;
